packages/oslo-generator-json-webuniversum/lib/JsonWebuniversumGenerationService.ts

```diff
@@ -0,0 +1,197 @@
+
+    // Sort entities
+    sortWebuniversumObjects(classes, this.configuration.language);
+    classes.forEach((classObject: WebuniversumObject) => sortWebuniversumObjects(classObject.properties || [], this.configuration.language));
+
+    sortWebuniversumObjects(datatypes, this.configuration.language);
+    datatypes.forEach((datatypeObject: WebuniversumObject) => sortWebuniversumObjects(datatypeObject.properties || [], this.configuration.language));
+
+    const baseURI = this.getBaseURI();
+
+    const template = { baseURI, classes: classes, dataTypes: datatypes };
+    await writeFile(this.configuration.output, JSON.stringify(template, null, 2), 'utf-8');
+  }
+
+  private getBaseURI(): string {
+    const packageSubject: RDF.Term = <RDF.Term>this.store.findQuad(null, ns.rdf('type'), ns.oslo('Package'))?.subject;
+
+    if (!packageSubject) {
+      throw new Error(`Unable to find the subject for the package.`);
+    }
+
+    const baseURIObject: RDF.Literal | undefined = <RDF.Literal | undefined>this.store.findObject(packageSubject, ns.oslo('baseURI'));
+
+    if (!baseURIObject) {
+      throw new Error(`Unable to find the baseURI for the package.`);
+    }
+
+    return baseURIObject.value;
+  }
+
+  private async generateEntityData(entity: RDF.NamedNode, includeProperties: boolean = true): Promise<WebuniversumObject> {
+    const assignedURI: RDF.NamedNode | undefined = this.store.getAssignedUri(entity);
+    if (!assignedURI) {
+      throw new Error(`Unable to find the assigned URI for entity ${entity.value}.`);
+    }
+
+    const fetchProperty = (fetchFunction: (subjectId: RDF.NamedNode, store: QuadStore, language: string) => RDF.Literal | undefined, subject: RDF.NamedNode) =>
+      fetchFunction(subject, this.store, this.configuration.language)?.value;
+
+    const vocabularyLabel: string | undefined = fetchProperty(getVocabularyLabel, entity);
+    const applicationProfileLabel: string | undefined = fetchProperty(getApplicationProfileLabel, entity);
+    const vocabularyDefinition: string | undefined = fetchProperty(getVocabularyDefinition, entity);
+    const applicationProfileDefinition: string | undefined = fetchProperty(getApplicationProfileDefinition, entity);
+    const vocabularyUsageNote: string | undefined = fetchProperty(getVocabularyUsageNote, entity);
+    const applicationProfileUsageNote: string | undefined = fetchProperty(getApplicationProfileUsageNote, entity);
+
+    const parentsIds: RDF.NamedNode[] = includeProperties ? this.store.getParentsOfClass(entity) : this.store.getParentOfProperty(entity) !== undefined ? [this.store.getParentOfProperty(entity)!] : [];
+    const parentObjects: Pick<WebuniversumObject, 'id' | 'vocabularyLabel' | 'applicationProfileLabel'>[] = parentsIds.map(parentId => this.createParentObject(parentId));
+
+    const scope: string | undefined = this.store.getScope(entity)?.value;
+
+    const entityData: WebuniversumObject = {
+      id: assignedURI.value,
+      ...vocabularyLabel && {
```

Comment:
I'm such a fan of the spread-operator. If I'm writing code in a non-js language, I always miss it 😆